bin/improver-weighted-blending

```diff
@@ -122,82 +202,93 @@ def main():
-    # used for this, because it would make it impossible to tell whether a
-    # a value was user inputted (which is needed for wrong_args_error calls).
-    if args.slope:
-        slope = args.slope
-    else:
-        slope = 0.0
-    if args.cval:
-        cval = args.cval
-    else:
-        cval = 0.85
-    if args.coord_adj:
-        args.coord_adj = eval(args.coord_adj)
 
+    # if the linear weights method is called with non-linear args or vice
+    # versa, exit with error
     if (args.wts_calc_method == "linear") and args.cval:
         parser.wrong_args_error('cval', 'linear')
-    elif ((args.wts_calc_method == "nonlinear") and
-          np.any([args.y0val, args.ynval, args.slope])):
-        parser.wrong_args_error('y0val, slope, ynval', 'non-linear')
+    if ((args.wts_calc_method == "nonlinear") and np.any([args.y0val,
+                                                          args.ynval])):
+        parser.wrong_args_error('y0val, ynval', 'non-linear')
+    if (args.wts_calc_method == "dict") and not args.wts_dict:
+        parser.error('Dictionary is required if --wts_calc_method="dict"')
 
-    if args.coordinate == 'time':
+    # temporary error - remove once this method is fully developed
+    if args.wts_calc_method == "mask":
+        parser.error('Calculation of weights from a cube mask is not yet '
+                     'supported')
+
+    # set blending coordinate units
+    if "time" in args.coordinate:
         coord_unit = Unit(args.coordinate_unit, args.calendar)
     elif args.coordinate_unit != 'hours since 1970-01-01 00:00:00.':
         coord_unit = args.coordinate_unit
     else:
         coord_unit = 'no_unit'
 
-    # Read in cubes and merge together if more than one found.
-    cubelist = iris.cube.CubeList([])
-    for ifile in args.input_filepaths:
-        cube_in = load_cube(ifile)
-        cubelist.append(cube_in)
-    if len(cubelist) == 1:
-        cube = cubelist[0]
-    else:
-        cube = merge_cubes(cubelist)
+    # load cubes to be blended
+    cubelist = load_cubelist(args.input_filepaths)
 
-    if args.wts_calc_method == "linear":
-        weights = ChooseDefaultWeightsLinear(
-            args.y0val, slope, args.ynval).process(
-                cube, args.coordinate, args.coord_exp_val, coord_unit,
-                args.wts_redistrib_method)
+    # determine whether or not to equalise forecast periods for model
+    # blending weights calculation
+    weighting_coord = (args.weighting_coord if args.weighting_coord
+                       else "forecast_period")
 
+    # ensure time coordinates are blendable
+    rationalise_blend_time_coords(
+        cubelist, args.coordinate, cycletime=args.cycletime,
+        weighting_coord=weighting_coord)
+
+    # merge cube list
+    cube = merge_cubes(cubelist)
+
+    # if the coord for blending does not exist or has only one value,
+    # update metadata only
+    coord_names = [coord.name() for coord in cube.coords()]
```

Comment:
Should we be doing something different if the coordinate doesn't exist, because the user may have made a mistake, compared to when the coordinate does exist but only has one point?

Reply:
Good point.  I think we want to raise a warning rather than an error if the coordinate doesn't exist here, so will add that, but am open to other viewpoints.